dexios/src/domain/erase.rs

```diff
@@ -0,0 +1,130 @@
+use rand::RngCore;
+use std::cell::RefCell;
+use std::fmt;
+use std::io::{Seek, Write};
+
+const BLOCK_SIZE: usize = 512;
+
+#[derive(Debug)]
+pub enum Error {
+    ResetCursorPosition,
+    OverwriteWithRandomBytes,
+    OverwriteWithZeros,
+    FlushFile,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Error::*;
+        match self {
+            ResetCursorPosition => f.write_str("Unable to reset cursor position"),
+            OverwriteWithRandomBytes => f.write_str("Unable to overwrite with random bytes"),
+            OverwriteWithZeros => f.write_str("Unable to overwrite with zeros"),
+            FlushFile => f.write_str("Unable to flush"),
+        }
+    }
+}
+
+impl std::error::Error for Error {}
+
+pub struct Request<W: Write + Seek> {
```

Comment:
@brxken128 it also gives us the ability to erase the file through an ssh connection or something like that) It would be a great feature!